src/chat.cpp

```diff
@@ -508,6 +520,18 @@ void CHAT_Tick( void )
 			players[i].lIgnoreChatTicks = -1;
 		}
 	}
+
+	// [AK] Reset the chat cursor's ticker if it goes too high.
+	if (( g_ulChatMode != CHATMODE_NONE ) && ( ++g_ulChatTicker >= TICRATE ))
+		g_ulChatTicker = 0;
+
+	// [AK] If we're typing a chat message to another player, we must constantly check if
+	// they're in the game. If not, cancel the message entirely.
+	if ( g_ulChatMode == CHATMODE_PRIVATE_SEND )
+	{
+		if (( g_ulChatPlayer != MAXPLAYERS ) && ( PLAYER_IsValidPlayer( g_ulChatPlayer ) == false ))
```

Comment:
Shouldn't this should be using `chat_IsPlayerValidReceiver` instead of `PLAYER_IsValidPlayer` now?

Reply:
I must've overlooked this line, sorry about that.